public/app/plugins/datasource/elasticsearch/datasource.test.ts

```diff
@@ -58,73 +72,68 @@ const createTimeRange = (from: DateTime, to: DateTime): TimeRange => ({
   },
 });
 
-interface Args {
-  data?: any;
+interface TestContext {
+  data?: Data;
   from?: string;
-  jsonData?: any;
+  jsonData?: Partial<ElasticsearchOptions>;
   database?: string;
-  mockImplementation?: (options: BackendSrvRequest) => Observable<FetchResponse>;
+  fetchImplementation?: (options: BackendSrvRequest) => Observable<FetchResponse>;
+}
+
+interface Data {
+  [key: string]: undefined | string | string[] | number | Data | Data[];
 }
 
 function getTestContext({
-  data = {},
+  data = { responses: [] },
   from = 'now-5m',
-  jsonData = {},
-  database = '[asd-]YYYY.MM.DD',
-  mockImplementation = undefined,
-}: Args = {}) {
-  jest.clearAllMocks();
-
+  jsonData,
+  database = '[test-]YYYY.MM.DD',
+  fetchImplementation = undefined,
+}: TestContext = {}) {
   const defaultMock = (options: BackendSrvRequest) => of(createFetchResponse(data));
 
   const fetchMock = jest.spyOn(backendSrv, 'fetch');
-  fetchMock.mockImplementation(mockImplementation ?? defaultMock);
+  fetchMock.mockImplementation(fetchImplementation ?? defaultMock);
 
-  const templateSrv: any = {
-    replace: jest.fn((text?: string) => {
+  const timeSrv = {
+    time: { from, to: 'now' },
+    timeRange: () => ({
+      from: dateMath.parse(timeSrv.time.from, false),
+      to: dateMath.parse(timeSrv.time.to, true),
+    }),
+    setTime: (time: RawTimeRange) => {
+      timeSrv.time = time;
+    },
+  } as TimeSrv;
+
+  const settings: Partial<DataSourceInstanceSettings<ElasticsearchOptions>> = { url: ELASTICSEARCH_MOCK_URL };
+  if (jsonData) {
```

Comment:
Another nit thing: why do we need that `if`?

Reply:
Probably a leftover from a previus change while dealing with types.